### ISSUE TITLE
bug(animations): Element rendered twice when animated

### ISSUE THREAD

**Issue by macjohnny** [angular/angular#26133]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Minimal reproduction of the problem with instructions
When triggering an animation of an element shown with `*ngIf` while an animation is already running, the element is rendered twice:
<img width="419" alt="bildschirmfoto 2018-09-27 um 12 31 37" src="https://user-images.githubusercontent.com/5589029/46140843-84727300-c252-11e8-90ba-caf24f9b915f.png">
The pink and the blue box should only be shown once. This happens when hitting the toggle button multiple times in the following example: [angular 7.0.0-beta.7](https://stackblitz.com/edit/angular-gitter-swjfyh?file=app/app.component.html)

This also occurs with angular 6.1.9: [angular 6.1.9](https://stackblitz.com/edit/angular-gitter-agjbr7?file=app%2Fapp.component.html)

This issue seems related to https://github.com/angular/angular/issues/23302

Note: at no time there exists more than one component instance, see this example: https://stackblitz.com/edit/angular-gitter-dp3cub
<img width="464" alt="bildschirmfoto 2018-09-27 um 12 58 12" src="https://user-images.githubusercontent.com/5589029/46141913-9a356780-c255-11e8-9fc9-dff38caeb0ba.png">

## What is the motivation / use case for changing the behavior?
It's a bug

## Environment

<pre><code>
Angular version: 7.0.0-beta.7

Browser:
- [x] Chrome (desktop) version 68
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version 62
- [x] Safari (desktop) version 12
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
</code></pre>

**Comment by macjohnny** [comment]
@matsko could you quickly have a look at this? Single elements with `*ngIf` appearing more than once  due to the animation is quite an issue...

**Comment by teesharp** [comment]
We just started seeing similar behavior in our app which started 2 days ago but we're running angular 5.2.5.

**Comment by simeyla** [comment]
I don't know what `severity1: confusing` means? Perhaps that means hard to duplicate.

Here's how :

I modified the tabs lazy loading demo [on stackblitz](https://stackblitz.com/edit/angular-bcnf44?file=app/tab-group-lazy-loaded-example.html).

The **only** thing I changed was adding a `div` element like this to each tab template.

    <ng-template matTabContent>
          <div>
             Content 2 - Loaded: {{getTimeLoaded(2) | date:'medium'}}
          </div>
     </ng-template> 

**You must have a device with a touch screen - or use a phone.**
**You likely won't be able to click fast enough with a mouse or trackpad.**

Then **touch** as fast as you can : 

* Tab 1
* Tab 2
* Tab 3
* Tab 1
* Tab 2 
* Tab 3 
* Repeat a few times.....

You'll see this for tab 2. 
Notably only the middle tab is affected, so that should aid in troubleshooting.

![image](https://user-images.githubusercontent.com/3656278/53296241-11990400-37c0-11e9-8ba9-fd2061962be5.png)

The user experience will vary depending upon your tabs content. If you have an absolutely positioned tab you may not notice, but with a flex layout it's a disaster.

**Comment by katyapavlenko** [comment]
This is solved by removing `*ngif` and animation directive from one element, but it there any hope of this being fixed in the future?

**Comment by gdinn** [comment]
I love angular but this is very frustrating.

**Comment by svidunovichromanv** [comment]
I have the same trouble.

**Comment by Mubramaj** [comment]
I have the same issue with angular 9.1

**Comment by Mubramaj** [comment]
any update on this ?

**Comment by Da13Harris** [comment]
We're also experiencing this bug in **Angular 9.1.9**. It's causing DOM elements to be duplicated when they have an `*ngIf` that's bound to the `enabled` property on a FormControl, and the same element is bound to the following `[@fadeIn]` animation:

```
...
animations: [
    trigger('fadeIn', [
        transition(':enter', [
            style({ opacity: 0 }),
            animate('300ms ease-in', style({ opacity: 1 }))
        ])
    ])
]
...
```

In our case, we're uncertain, but pretty sure the issue didn't occur in previous versions of Angular.

**Comment by Da13Harris** [comment]
This issue seems to be related to [#35270](https://github.com/angular/angular/issues/35270)

**Comment by BeejeeDS** [comment]
Having the same issue even when using [@.disabled].
Could this be related to my other issue #35825

**Comment by picheli20** [comment]
We were facing the same problem on at my company - after some investigation I've noticed that the problem was due to `ng-content` with `*ngIf`, for example:
```
<div *ngIf="display">
    <ng-content></ng-content>
</div>
```

Even the `ng-content` being deleted from the DOM, it wasn't really updated (or cached) on Virtual DOM (I believe), one solution that worked for us was to have a `*ngIf` in the component which `ng-content` is, and trigger it when the animation is done `(@collapseBounce.done)="onAnimationFinish()`

**Comment by luizgrs** [comment]
still a problem in v16, moving the animation trigger to a child of *ngIf fixed it to me and kept animation working

**Comment by aeslinger0** [comment]
Still a problem in v17. @luizgrs thanks for the work-around. I'm still seeing an intermittent unexpected playing of the animation by putting the animation on a child of the *ngIf (probably a race condition with router events setting the ngIf), but it avoids the duplicate element, so it's tolerable for me.

**Comment by junaidahmed501** [comment]
> still a problem in v16, moving the animation trigger to a child of *ngIf fixed it to me and kept animation working

Yes, this indeed worked for me as well in v17. I had to move the animation to the direct-child of my element that has ngIf. Thanks

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
